doc/openshift-guide.md

```diff
@@ -0,0 +1,205 @@
+
+Navigate to the **Dashboard > API** tab, if you have more than one API in your account, select the API you want to manage with the API Gateway. Select the **Integration** link at top-left.
+
+If you're setting this up for the first time, you'll need to test to confirm that your private (unmanaged) API is working before proceeding. If you've already configured your API and sent test traffic, feel free to skip this step.
+
+<img src="https://support.3scale.net/images/screenshots/guides-openshift-private-base-url.png" alt="Private base URL">
+
+In the screenshot above, this API is configured to use the 3scale provided Echo API to help you get started. You can use this or configure the API to refer to your real API's _Private Base URL_.
+
+Test your private (unmanaged) API is working using this _curl_ command:
+    
+    curl https://echo-api.3scale.net:443/
+
+You should get a response similar to this:
+
+    {
+      "method": "GET",
+      "path": "/",
+      "args": "",
+      "body": "",
+      "headers": {
+        "HTTP_VERSION": "HTTP/1.1",
+        "HTTP_HOST": "echo-api.3scale.net",
+        "HTTP_USER_AGENT": "curl/7.43.0",
+        "HTTP_ACCEPT": "*/*",
+        "HTTP_X_FORWARDED_FOR": "10.1.0.154",
+        "HTTP_CONNECTION": "close"
+    }
+
+Once you've confirmed that your API is working, scroll down to the bottom of the **Staging** section where you can see a sample _curl_ command:
+
+    curl "https://XXX.staging.apicast.io:443/?user_key=YOUR_USER_KEY"
+
+**Note**: Note down *YOUR_USER_KEY* to use later to authenticate to your managed API.
+
+Next you need to set your API to be a Self-managed Gateway, required to run the gateway on OpenShift. Select the **edit integration settings** link at top-right on the Integration page you are on. Choose **NGINX Self-managed Gateway** and save it by clicking **Update Service**.
+
+You should now see a section **Production: Self-managed Gateway** at the bottom of the Integration page. You need to add a _Public Base URL_ appropriate for this API or Service, that will be the URL of your API gateway. In case you have multiple API services, you will need to set this _Public Base URL_ appropriately for each service. In this example we will use `http://gateway.openshift.demo:80` as _Public Base URL_, but typically it will be something like `https://api.yourdomain.com:443`, on the domain that you manage (`yourdomain.com`).
+
+### Setup OpenShift
+
+1. Download required tools and files
+  - Go to <a href="https://www.openshift.org/vm/">https://www.openshift.org/vm/ _DOWNLOADS_ section.
+  - Download and install the required tools <a href="https://www.vagrantup.com/downloads.html" target="_blank">Vagrant</a> and <a href="https://www.virtualbox.org/wiki/Downloads" target="_blank">VirtualBox</a>.
```

Comment:
Do we really want to go the vagrant route? There is `oc cluster up` to start cluster in Docker. Either natively or provisioned by docker machine.